src/data_generators/test_lhs_synthetic_data_design_config.py

```diff
@@ -0,0 +1,111 @@
+from wfa_planning_evaluation_framework.data_generators.homogeneous_impression_generator import HomogeneousImpressionGenerator
+from wfa_planning_evaluation_framework.data_generators.independent_overlap_data_set import IndependentOverlapDataSet
+from wfa_planning_evaluation_framework.data_generators.synthetic_data_design_config import SyntheticDataDesignConfig
+
+NUM_PUBS = "num_publishers"
+LARGEST_PUB_SIZE = "largest_publisher_size"
+PUB_RATIO = "largest_to_smallest_publisher_ratio"
+IMG_GEN_PARAMS = "impression_generator_params"
+PRICE_GEN_PARAMS = "pricing_generator_params"
+OVERLAP_GEN_PARAMS = "overlap_generator_params"
+
+RANDOMIZATION_NEEDED_GEN_PARAMS = [IMG_GEN_PARAMS, OVERLAP_GEN_PARAMS]
+
+NUM_SAMPLES_FOR_LHS = 3
+NUM_RANDOM_REPLICAS = 2
+
+class TestLHSSyntheticDataDesignConfig(SyntheticDataDesignConfig):
+  """Generates a DataDesign using LHS from a synthetic grid."""
+
+  @classmethod
+  def get_lhs_design(cls, variable_grid_axes):
+    levels = [
+        len(variable_levels)
+        for variable_levels in list(variable_grid_axes.values())
+    ]
+    design = lhs(
+        n=len(levels), samples=NUM_SAMPLES_FOR_LHS, criterion="maximin")
+    return (design * np.array(levels)).astype("int32")
+
+  @classmethod
+  def get_grid_axes(cls):
+    grid_axes = OrderedDict()
+    grid_axes[NUM_PUBS] = [1, 2, 5]
+    grid_axes[LARGEST_PUB_SIZE] = [100, 1000]
+    grid_axes[PUB_RATIO] = [1, 0.5, 0.3, 0.1]
+    grid_axes[IMG_GEN_PARAMS] = [
+        GeneratorParameters(
+            generator=HomogeneousImpressionGenerator,
+            params={"poisson_lambda": 2}),
+        GeneratorParameters(
+            generator=HomogeneousImpressionGenerator,
+            params={"poisson_lambda": 5})
+    ]
+    grid_axes[PRICE_GEN_PARAMS] = [
+        GeneratorParameters(
+            generator=FixedPriceGenerator, params={"cost_per_impression": 0.1})
+    ]
+    grid_axes[OVERLAP_GEN_PARAMS] = [
+        GeneratorParameters(generator=IndependentOverlapDataSet, params={})
+    ]
+    return grid_axes
+
+  @classmethod
+  def get_params_dict(cls, lhs_trial: List[int], grid_axes: OrderedDict):
+    keys = list(grid_axes.keys())
+    values = list(grid_axes.values())
+    params_dict = {}
+    for param in range(len(lhs_trial)):
+      params_dict[keys[param]] = values[param][lhs_trial[param]]
+    return params_dict
+
+  @classmethod
+  def get_data_set_params_list(
+      cls, random_generator: Generator) -> List[DataSetParameters]:
+    grid_axes = cls.get_grid_axes()
+    lhs_design = cls.get_lhs_design(grid_axes)
+
+    return [
+        cls.get_data_set_params(
+            cls.get_params_dict(lhs_trial, grid_axes), random_generator)
+        for lhs_trial in lhs_design
+        for i in range(NUM_RANDOM_REPLICAS)
+    ]
+
+  @classmethod
+  def get_data_set_params(cls, params_dict,
+                          random_generator: Generator) -> DataSetParameters:
```

Comment:
Describe the input output, better with an example. Or add a test to show example